--- ipw2100-1.2.2/ipw2100.c	2008-03-02 21:56:02.755475954 +0100
+++ ipw2100-1.2.2/ipw2100.c~	2008-03-02 21:55:31.000000000 +0100
@@ -6383,9 +6383,9 @@ static int ipw2100_pci_init_one(struct p
 	dev->irq = pci_dev->irq;
 
 	IPW_DEBUG_INFO("Attempting to register device...\n");
-
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24)
 	SET_MODULE_OWNER(dev);
-
+#endif
 	printk(KERN_INFO DRV_NAME
 	       ": Detected Intel PRO/Wireless 2100 Network Connection\n");
 
