diff -u ipw2100-0.46_3.orig/ieee80211.c ipw2100-0.46_3/ieee80211.c
--- ipw2100-0.46_3.orig/ieee80211.c	2004-06-08 02:42:57.000000000 +0200
+++ ipw2100-0.46_3/ieee80211.c	2004-07-04 00:12:22.195495536 +0200
@@ -29,7 +29,6 @@
   Intel Corporation, 5200 N.E. Elam Young Parkway, Hillsboro, OR 97124-6497
 
 *******************************************************************************/
-#include <linux/802_11.h>
 #include <linux/compiler.h>
 #include <linux/config.h>
 #include <linux/errno.h>
diff -u ipw2100-0.46_3.orig/ieee80211.h ipw2100-0.46_3/ieee80211.h
--- ipw2100-0.46_3.orig/ieee80211.h	2004-06-08 02:42:57.000000000 +0200
+++ ipw2100-0.46_3/ieee80211.h	2004-07-04 00:12:18.952988472 +0200
@@ -43,7 +43,6 @@
 
 /* IEEE 802.11 defines */
 
-#include <linux/802_11.h>
 #define SNAP_SIZE sizeof(struct ieee_802_11_snap_header)
 
 
diff -u ipw2100-0.46_3.orig/ieee80211_rx.c ipw2100-0.46_3/ieee80211_rx.c
--- ipw2100-0.46_3.orig/ieee80211_rx.c	2004-06-08 02:42:57.000000000 +0200
+++ ipw2100-0.46_3/ieee80211_rx.c	2004-07-04 00:12:27.911626552 +0200
@@ -12,7 +12,6 @@
  * published by the Free Software Foundation. See README and COPYING for
  * more details.
  */
-#include <linux/802_11.h>
 #include <linux/compiler.h>
 #include <linux/config.h>
 #include <linux/errno.h>
diff -u ipw2100-0.46_3.orig/ieee80211_tx.c ipw2100-0.46_3/ieee80211_tx.c
--- ipw2100-0.46_3.orig/ieee80211_tx.c	2004-06-08 02:42:57.000000000 +0200
+++ ipw2100-0.46_3/ieee80211_tx.c	2004-07-04 00:12:30.308262208 +0200
@@ -24,7 +24,6 @@
   Intel Corporation, 5200 N.E. Elam Young Parkway, Hillsboro, OR 97124-6497
 
 ******************************************************************************/
-#include <linux/802_11.h>
 #include <linux/compiler.h>
 #include <linux/config.h>
 #include <linux/errno.h>
diff -u ipw2100-0.46_3.orig/ipw2100_fw.c ipw2100-0.46_3/ipw2100_fw.c
--- ipw2100-0.46_3.orig/ipw2100_fw.c	2004-06-08 02:42:57.000000000 +0200
+++ ipw2100-0.46_3/ipw2100_fw.c	2004-07-04 00:12:43.810209600 +0200
@@ -37,7 +37,6 @@
 
 ******************************************************************************/
 
-#include <linux/802_11.h>
 #include <linux/compiler.h>
 #include <linux/config.h>
 #include <linux/errno.h>
diff -u ipw2100-0.46_3.orig/ipw2100_main.c ipw2100-0.46_3/ipw2100_main.c
--- ipw2100-0.46_3.orig/ipw2100_main.c	2004-07-04 00:05:34.000000000 +0200
+++ ipw2100-0.46_3/ipw2100_main.c	2004-07-04 00:17:30.420638192 +0200
@@ -141,7 +141,6 @@
 
 */
 
-#include <linux/802_11.h>
 #include <linux/compiler.h>
 #include <linux/config.h>
 #include <linux/errno.h>
@@ -1531,17 +1530,6 @@
 	unsigned long flags;
 	unsigned long delay;
 	spin_lock_irqsave(&priv->low_lock, flags);
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,0)
-	if (priv->reset_backoff) {
-		delay = priv->reset_backoff * HZ;
-		spin_unlock_irqrestore(&priv->low_lock, flags);
-
-		set_current_state(TASK_UNINTERRUPTIBLE);
-		schedule_timeout(delay);
-
-		spin_lock_irqsave(&priv->low_lock, flags);
-	}
-#endif
 	printk(KERN_INFO DRV_NAME ": %s: Restarting adapter.\n", 
 	       priv->ndev->name);
 	priv->resets++;
@@ -1972,15 +1960,15 @@
 	char *msg = "UNKNOWN";
 	
 	switch (WLAN_FC_GET_STYPE(header->frame_control)) {
-	case ieee_802_11_frame_subtype_Association_Resp:
+	case WLAN_FC_STYPE_ASSOC_RESP:
 		msg = "ASSOCIATION RESPONSE";
 		break;
 
-	case ieee_802_11_frame_subtype_Reassociation_Resp:
+	case WLAN_FC_STYPE_REASSOC_RESP:
 		msg = "REASSOCIATION RESPONSE";
 		break;
 
-	case ieee_802_11_frame_subtype_Probe_Resp:
+	case WLAN_FC_STYPE_PROBE_RESP:
 		msg = "PROBE RESPONSE";
 		IPW2100_DEBUG_SCAN("Probe response\n");
 		isr_process_probe_response(
@@ -1988,7 +1976,7 @@
 			status->rssi);
 		break;
 
-	case ieee_802_11_frame_subtype_Beacon:
+	case WLAN_FC_STYPE_BEACON:
 		msg = "BEACON";
 		IPW2100_DEBUG_SCAN("Beacon\n");
 		isr_process_probe_response(
