diff -u clean/ipw2100-0.46_3/ieee80211.c ipw2100-0.46_3/ieee80211.c
--- clean/ipw2100-0.46_3/ieee80211.c	2004-06-07 17:42:57.000000000 -0700
+++ ipw2100-0.46_3/ieee80211.c	2004-06-21 18:46:31.000000000 -0700
@@ -29,7 +29,7 @@
   Intel Corporation, 5200 N.E. Elam Young Parkway, Hillsboro, OR 97124-6497
 
 *******************************************************************************/
-#include <linux/802_11.h>
+
 #include <linux/compiler.h>
 #include <linux/config.h>
 #include <linux/errno.h>
@@ -51,6 +51,7 @@
 #include <linux/etherdevice.h>
 #include <asm/uaccess.h>
 
+#include "ieee802_11.h"
 #include "ieee80211.h"
 
 #ifndef CONFIG_IPW2100_NOWEP
diff -u clean/ipw2100-0.46_3/ieee80211.h ipw2100-0.46_3/ieee80211.h
--- clean/ipw2100-0.46_3/ieee80211.h	2004-06-07 17:42:57.000000000 -0700
+++ ipw2100-0.46_3/ieee80211.h	2004-06-21 18:58:16.000000000 -0700
@@ -43,9 +43,19 @@
 
 /* IEEE 802.11 defines */
 
-#include <linux/802_11.h>
+#include "ieee802_11.h"
 #define SNAP_SIZE sizeof(struct ieee_802_11_snap_header)
 
+#define P80211_OUI_LEN 3
+
+struct ieee_802_11_snap_header {
+
+        u8    dsap;   /* always 0xAA */
+        u8    ssap;   /* always 0xAA */
+        u8    ctrl;   /* always 0x03 */
+        u8    oui[P80211_OUI_LEN];    /* organizational universal id */
+
+} __attribute__ ((packed));
 
 #define WLAN_FC_PVER (BIT(1) | BIT(0))
 #define WLAN_FC_TODS BIT(8)
@@ -59,7 +69,7 @@
 
 #define WLAN_FC_GET_TYPE(fc) (((fc) & (BIT(3) | BIT(2))) >> 2)
 #define WLAN_FC_GET_STYPE(fc) \
-	(((fc) & (BIT(7) | BIT(6) | BIT(5) | BIT(4))) >> 4)
+	(((fc) & (BIT(7) | BIT(6) | BIT(5) | BIT(4))) >> 3)
 
 #define WLAN_GET_SEQ_FRAG(seq) ((seq) & (BIT(3) | BIT(2) | BIT(1) | BIT(0)))
 #define WLAN_GET_SEQ_SEQ(seq) \
diff -u clean/ipw2100-0.46_3/ieee80211_rx.c ipw2100-0.46_3/ieee80211_rx.c
--- clean/ipw2100-0.46_3/ieee80211_rx.c	2004-06-07 17:42:57.000000000 -0700
+++ ipw2100-0.46_3/ieee80211_rx.c	2004-06-21 18:55:33.000000000 -0700
@@ -12,7 +12,7 @@
  * published by the Free Software Foundation. See README and COPYING for
  * more details.
  */
-#include <linux/802_11.h>
+
 #include <linux/compiler.h>
 #include <linux/config.h>
 #include <linux/errno.h>
diff -u clean/ipw2100-0.46_3/ieee80211_tx.c ipw2100-0.46_3/ieee80211_tx.c
--- clean/ipw2100-0.46_3/ieee80211_tx.c	2004-06-07 17:42:57.000000000 -0700
+++ ipw2100-0.46_3/ieee80211_tx.c	2004-06-21 18:52:56.000000000 -0700
@@ -24,7 +24,6 @@
   Intel Corporation, 5200 N.E. Elam Young Parkway, Hillsboro, OR 97124-6497
 
 ******************************************************************************/
-#include <linux/802_11.h>
 #include <linux/compiler.h>
 #include <linux/config.h>
 #include <linux/errno.h>
diff -u clean/ipw2100-0.46_3/ipw2100_fw.c ipw2100-0.46_3/ipw2100_fw.c
--- clean/ipw2100-0.46_3/ipw2100_fw.c	2004-06-07 17:42:57.000000000 -0700
+++ ipw2100-0.46_3/ipw2100_fw.c	2004-06-21 18:51:47.000000000 -0700
@@ -37,7 +37,6 @@
 
 ******************************************************************************/
 
-#include <linux/802_11.h>
 #include <linux/compiler.h>
 #include <linux/config.h>
 #include <linux/errno.h>
diff -u clean/ipw2100-0.46_3/ipw2100_main.c ipw2100-0.46_3/ipw2100_main.c
--- clean/ipw2100-0.46_3/ipw2100_main.c	2004-06-07 17:42:57.000000000 -0700
+++ ipw2100-0.46_3/ipw2100_main.c	2004-06-21 17:57:22.000000000 -0700
@@ -141,7 +141,6 @@
 
 */
 
-#include <linux/802_11.h>
 #include <linux/compiler.h>
 #include <linux/config.h>
 #include <linux/errno.h>
@@ -163,6 +162,7 @@
 #include <linux/time.h>
 #include <asm/uaccess.h>
 
+#include "ieee802_11.h"
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,0)
 #include <linux/sched.h>
@@ -1956,15 +1956,15 @@
 	char *msg = "UNKNOWN";
 	
 	switch (WLAN_FC_GET_STYPE(header->frame_control)) {
-	case ieee_802_11_frame_subtype_Association_Resp:
+	case IEEE802_11_STYPE_ASSOC_RESP:
 		msg = "ASSOCIATION RESPONSE";
 		break;
 
-	case ieee_802_11_frame_subtype_Reassociation_Resp:
+	case IEEE802_11_STYPE_REASSOC_RESP:
 		msg = "REASSOCIATION RESPONSE";
 		break;
 
-	case ieee_802_11_frame_subtype_Probe_Resp:
+	case IEEE802_11_STYPE_PROBE_RESP:
 		msg = "PROBE RESPONSE";
 		IPW2100_DEBUG_SCAN("Probe response\n");
 		isr_process_probe_response(
@@ -1972,7 +1972,7 @@
 			status->rssi);
 		break;
 
-	case ieee_802_11_frame_subtype_Beacon:
+	case IEEE802_11_STYPE_BEACON:
 		msg = "BEACON";
 		IPW2100_DEBUG_SCAN("Beacon\n");
 		isr_process_probe_response(
